src/app/shared/pipes/auth-provider-name.pipe.ts

```diff
@@ -0,0 +1,30 @@
+import {Injectable, Pipe, PipeTransform} from '@angular/core'
+import {AuthProvider} from "../../preference/state/preference.store"
+
+@Injectable({
+  providedIn: 'root',
+})
+@Pipe({
+  name: 'authProviderName',
+})
+export class AuthProviderNamePipe implements PipeTransform {
+  constructor() {
+  }
+
+  transform(value: '' | AuthProvider | null | undefined): string {
+    if (!value) {
+      return ''
+    }
+
+    switch (value as AuthProvider) {
+      case AuthProvider.METAMASK:
+        return 'Metamask'
+      case AuthProvider.WALLET_CONNECT:
+        return 'Wallet Connect'
+      case AuthProvider.VENLY:
+        return 'Venly'
+      default:
+        return ''
```

Comment:
Maybe return null here or throw an exception? It should fail if unspecified provider is used in template.

Reply:
Good point, I will throw an exception.